crates/db_connection_pool/src/dbconnection/snowflakeconn.rs

```diff
@@ -145,3 +169,69 @@ impl<'a> AsyncDbConnection<Arc<SnowflakeApi>, &'a (dyn Sync)> for SnowflakeConne
 fn to_execution_error(e: impl Into<Box<dyn std::error::Error>>) -> DataFusionError {
     DataFusionError::Execution(format!("{}", e.into()).to_string())
 }
+
+/// Converts `Snowflake` specific types to standard Arrow types.
+///
+/// # Errors
+///
+/// Returns an error if there is a failure in converting Snowflake to Arrow types.
+pub fn snowflake_schema_cast(record_batch: &RecordBatch) -> Result<RecordBatch, Error> {
+    let mut fields = Vec::new();
+    let mut columns = Vec::new();
+
+    for (idx, field) in record_batch.schema().fields().iter().enumerate() {
+        let column = record_batch.column(idx);
+        if let Some(sf_logical_type) = field.metadata().get("logicalType") {
+            if sf_logical_type.to_lowercase().as_str() == "timestamp_ntz" {
+                fields.push(Arc::new(Field::new(
+                    field.name(),
+                    DataType::Timestamp(TimeUnit::Millisecond, None),
+                    field.is_nullable(),
+                )));
+                columns.push(cast_sf_timestamp_ntz_to_arrow_timestamp(column)?);
+                continue;
+            }
+        }
+        fields.push(Arc::clone(field));
+        columns.push(Arc::clone(column));
+    }
+
+    let schema = Arc::new(Schema::new(fields));
+    RecordBatch::try_new(schema, columns).context(FailedToCreateRecordBatchSnafu)
+}
+
+fn cast_sf_timestamp_ntz_to_arrow_timestamp(column: &ArrayRef) -> Result<ArrayRef, Error> {
```

Comment:
Let's add some tests for this function

Reply:
Added unit tests for `cast_sf_timestamp_ntz_to_arrow_timestamp`